chain/client/src/client.rs

```diff
@@ -676,6 +682,18 @@ impl Client {
             }
         };
 
+        // Process stored partial encoded chunks
+        let next_height = block.header.inner_lite.height + 1;
+        let mut partial_encoded_chunks =
+            self.shards_mgr.get_stored_partial_encoded_chunks(next_height);
+        for (_shard_id, partial_encoded_chunk) in partial_encoded_chunks.drain() {
+            // We will request chunks later if cannot process them here
+            let accepted_blocks = self.process_partial_encoded_chunk(partial_encoded_chunk);
+            if accepted_blocks.is_ok() {
+                debug_assert!(accepted_blocks.unwrap().len() == 0);
```

Comment:
Actually now that I thikn about it, we can have accepted blocks as a result of processing chunks here. Any block that is in the `blocks_with_missing_chunks` which doesn't have any chunks for which we track shards will be unblocked here.

So what we should do instead is move this entire logic to the very end of `on_block_accepted`, and for each element of the `accepted_blocks` call `on_block_accepted` again.

I would expect `cross_shard_tx_with_validator_rotation` to actually catch this `debug_assert`, try running it end-to-end couple times?